### ISSUE TITLE
Search in caret mode

### ISSUE THREAD

**Issue by rcorre** [The-Compiler/qutebrowser#1402]
I'd like to yank some text between "foo" and "bar", presumably by doing:

`/foo<cr>` to select "foo"
`v` to enter caret mode
`/bar<cr>` to move the caret _and extend my selection_ to "bar" 
`y` to yank the selected text.

Unfortunately, search doesn't work in caret mode

**Comment by The-Compiler** [comment]
The first step towards making this work would be `:bind --mode caret / set-cmd-text /` - unfortunately, entering search mode also drops out of caret mode.

No idea how straightforward it'd be to fix that...

cc @artur-shaik

**Comment by artur-shaik** [comment]
Looks like it requires something like filo stack of key modes, to give abillity to return to visual mode after command finished.

**Comment by The-Compiler** [comment]
Yeah, I thought so too. Actually that used to be the case, but I ditched the mode stack shortly after the first release as it was a troublemaker: 03ac8874ff3036069c72e96f6268235a34b26ea2

**Comment by artur-shaik** [comment]
Yeah, actually now I think that stack it too complicated approach. May be it is better to just save 'backmode' for command mode only.
